### DISCUSSION TITLE
Feedback needed on new UI - Material Relationships & Inheritance

### DISCUSSION THREAD

**Discussion by womanyoyoyo-adsk** [Autodesk/maya-usd#2744] [Show and tell]
Hey folks,

Our team is working on some new features for MayaUSD and I wanted to get some feedback from all of you about the designs currently being worked on.

Please watch the two attached videos containing the design options for showing the material relationships and inheritance attributes for a prim.

As you’d be able to see in the videos, in the Attribute Editor, under USD Prims, we are creating a new section **Materials** to show which prim has which material assigned to it, and show inheritance and strength.

Here’s a few questions I would like to hear your thoughts on:

1. Which of the two design options works better to show the strength of the materials? The “Strength” dropdown? or the “Override Children Materials” checkbox?
2. How important do you think USD Material Strengths feature will be for your workflow?
3. Does the design show all the information you would like to see about material relationships? Is the information clear?
4. Any other feedback or thoughts you may have about the design.
5. How do you think you would you use this feature? Any interesting examples?

Thank you, and looking forward to hearing from all of you.

https://user-images.githubusercontent.com/49446314/203869366-95da914e-8328-414e-882c-8591fcc8b59b.mov

https://user-images.githubusercontent.com/49446314/203869439-ca423e89-75a2-46db-a740-928e5d2169da.mov

**Comment by oumad** [comment]
Hi, thanks for the detailed videos.

1.  have preference for the first design, it's the correct USD terminology and it's more consistent with what we would find in other packages (omniverse etc..), and it's the most explicit too.
2. I find the material strength feature more flexible and useful than the default maya's by shape shading, it offers way more possibilities and ease for both artists and TDs in my view.
3.  I'm pretty pleased with the way we can easily access both the inherited material or the prim that it's assigned to, I believe there is much less risk of confusion comparing to how other packages I've seen implemented this.
4.  The icon that supposedly allows us to access the material graph, at first I thought it's there to break the connection, it's a common thing especially for windows users that the X in the right symbolizes closing or disgarding something. So I'd suggest using a different icon.
5.  This is very handy for an iterative workflow when shading assets with too many pieces, for example for a car asset, where you can select a body group and assign a car paint to it as the most dominant visually, then overtime shade the child pieces that might have other materials (plastic, chrome..).

We've been waiting ages to have this in maya ! keep it up.

**Reply by shajeeaijazi** [reply]
Mohamed, 

Thank you for the detailed answers, they're super helpful. 

Good feedback about the icon for graphing out - that's meant to be the LookdevX icon, hence the X, but definitely something to keep in mind.

Again, much appreciated.

Shajee Aijazi,
Sr. Lead UX Designer
LookdevX

**Reply by oumad** [reply]
I figured that was for LookdevX, in that case maybe you can try putting the logo inside a square, that can have 2 benefits : 
1- Make the X smaller, so less apparent as the main action.
2- It would be better understood that this leads to a different window as opposed to an action in the current field, kind of like the square we get next to many commands in maya : 
![image](https://user-images.githubusercontent.com/17836943/204037191-8a1e944e-bc1f-4c4c-acef-974229ee9057.png)
Also similar to the icon that take us to the assigned material or prim : 
![image](https://user-images.githubusercontent.com/17836943/204037291-77ef2cf4-3f00-4d46-a8e1-a6b12d6f0713.png)
or ones in the node editor : 
![image](https://user-images.githubusercontent.com/17836943/204037438-44fb75ae-afee-42d0-9479-2186ade3b9b0.png)

I hope this makes sense ?

**Comment by daviddeacon** [comment]
The implementation looks great, it looks like it's really easy to see where a material is being assigned or inherited from. I really like the jump to buttons so it's quick to navigate around the stage. I would probably go for option 1, option 2 probably makes more sense to the average Maya user but it's probably better to stick to the correct USD lingo to help with consistency.

PS, On the subject of design, please add a dedicated option to the outliner to expand the USD stage, so you don't have to have show shapes turned on. It's really annoying  having it on when also working with Maya rigs. I know it's a long requested feature.

**Comment by gelshoffNV** [comment]
I like the first one, not sure the prim that "owns" the assignment has to be immediately seen / accessible, to me it's more of a Maya node connection, where you see it's being driven by something, but you go elsewhere to dive in and actually see that detail. But maybe a small button to quick select the prim. I'm still new to USD, but I know there is collection based assignment....is the collection considered a prim? Curious if that concept would still work here if it's not. There also seems to be some difference in the way a collection assignment overrides the direct assignment, vs inheritance strength. It looks like USD may support collections with syntax like /pencils/*/eraser, the docs mention how Katana uses CEL and USD seems to understand that...you may want to consider non inheritance / direct assignment and how that may look with your UI.

**Reply by sharktacos** [reply]
Has this been implemented yet?

**Reply by shajeeaijazi** [reply]
@sharktacos This has not been implemented yet, but it's planned soon.

**Comment by sepux** [comment]
Def option 1 - I hope this design come soon

**Reply by shajeeaijazi** [reply]
@sepux thank you for sharing your preferred option - and yes, it's coming soon.
